qlib/ConnectionProvider.qm

```diff
@@ -366,34 +375,92 @@ string connstr = get_connection(str);
             @param n_urlh broken down URL hash (as returned by @ref Qore::parse_url())
             @param n_safe_url "safe" URL (password information removed); if not set this will be set automatically by calling @ref getSafeUrl()
             @param n_enabled enabled/disabled flag
+
+            @deprecated since Qore 0.9
          */
+        deprecated
         constructor(string n_name, string n_desc, string n_url, bool n_monitor, *hash n_opts, hash n_urlh, *string n_safe_url,
                     *bool n_enabled) {
-            name = n_name;
-            desc = n_desc;
-            url = n_url;
-            monitor = n_monitor;
-            safe_url = n_safe_url ? n_safe_url : AbstractConnection::getSafeUrl(n_urlh);
-            urlh = n_urlh;
-            orig_opts = n_opts;
-            n_opts = getDefaultOptions() + n_opts;
-            if (n_opts) {
+            hash attributes = {
+                "monitor": n_monitor ?? True,
+                "enabled": n_enabled ?? True,
+            };
+            constructorInit(n_name, n_desc, n_url, attributes, n_opts ?? {});
+        }
+
+        #! creates the AbstractConnection object
+        /** @param name the name of the connection
+            @param description connection description
+            @param url connection URL (potentially with password info)
+            @param attributes various attributes. See below
+            @param options connection options
+
+            @par Attributes
+
+            Attributes are special flags for given connection - stored as public
+            attributes of the object instance.
+            For now there are:
+                - \c enabled, default \c True
+                - \c monitor, default \c True
+
+            @par Safe URL
```

Comment:
why `@par Safe URL` is here, documenting constructor, if there is no arg like `safe_url` anymore? Shouldn't this be documented by the member or class itself?

Reply:
yes, done